### PR TITLE
Jormun: suspension of user account

### PR DESCRIPTION
**Pull request by lrocheWB**
depends of https://github.com/CanalTP/navitia/pull/1338

### CODE REVIEW

File: source/jormungandr/jormungandr/quota.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+from datetime import datetime
+
+def quota_control(func):
+    """
+    Decorator for quota control on every request
+    """
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        user = authentication.get_user(token=authentication.get_token())
+        if user.block_until and datetime.utcnow().date() <= user.block_until:
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/quota.py]: Not convinced about this.

I think we should at least encapsulate this but wouldn't it be better to integrate this into user.has_access ?

who will update this `block_until` ?

why store only a date ? don't you want a date time ?

File: source/jormungandr/jormungandr/interfaces/v1/StatedResource.py [hunk 2]
```diff
@@ -54,4 +58,4 @@ def wrapper(*args, **kwargs):
             if region:
                 stat_manager.register_regions([region])
             return result
-        return wrapper
+        return wrapper
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/interfaces/v1/StatedResource.py]: it's better if you add a line at the end of the file

File: source/jormungandr/jormungandr/quota.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+from datetime import datetime
+
+def quota_control(func):
+    """
+    Decorator for quota control on every request
+    """
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        user = authentication.get_user(token=authentication.get_token())
+        if user.block_until and datetime.utcnow() <= user.block_until:
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/quota.py]: I think this should at least be the responsability of the User class

File: source/jormungandr/jormungandr/quota.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            flask_restful.abort(429, message="Quota limit reached, please contact provider if you want to upgrade your current billing plan")
+
+        return func(*args, **kwargs)
+
+    return wrapper
```
Comment (antoine-de) [inline source/jormungandr/jormungandr/quota.py]: same minor remark, you should add a line

**Comment by Netmisa** [comment]
Jenkins could you test this please.

**Comment by Netmisa** [comment]
Jenkins, test this please

**Comment by antoine-de** [comment]
Jenkins, test this please

**Comment by Netmisa** [comment]
@lrocheWB 

This PR https://github.com/CanalTP/navitia/pull/1338 are merged.
Could you rebase dev branch please ?

**Comment by Netmisa** [comment]
I have an error 500

```
TypeError: can't compare offset-naive and offset-aware datetimes
```

**Comment by antoine-de** [comment]
The Jormungandr Unit Tests do not work. Can you check them ?

**Comment by lrocheWB** [comment]
@antoine-de should be good :)

File: source/navitiacommon/navitiacommon/models.py [hunk 5]
```diff
@@ -173,6 +173,13 @@ def has_access(self, instance_id, api_name):
 
         return query.count() > 0
 
+    def is_blocked(self, datetime_utc):
+        print self.block_until
```
Comment (Netmisa) [inline source/navitiacommon/navitiacommon/models.py]: Debug ?

File: source/navitiacommon/navitiacommon/models.py [hunk 6]
```diff
@@ -173,6 +173,13 @@ def has_access(self, instance_id, api_name):
...
+        print datetime_utc
```
Comment (Netmisa) [inline source/navitiacommon/navitiacommon/models.py]: Debug ?

File: source/jormungandr/jormungandr/quota.py [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+from datetime import datetime
+
+def quota_control(func):
+    """
+    Decorator for quota control on every request
+    """
+    @wraps(func)
+    def wrapper(*args, **kwargs):
+        user = authentication.get_user(token=authentication.get_token())
+        if user.is_blocked(datetime.utcnow()):
```
Comment (Netmisa) [inline source/jormungandr/jormungandr/quota.py]: When user send a bad token he have an error 500 HTTP
You need to update your condition here by this

```
if user is not None and user.is_blocked(datetime.utcnow()):
```

You also need to update your process order to check before this if user is authenticate.

**Comment by antoine-de** [comment]
Jenkins, test this please

**Pull request merged**